### DISCUSSION TITLE
Library internals overview?

### DISCUSSION THREAD

**Discussion by roccomoretti** [molstar/molstar#837] [General]
Is there a document somewhere which summarizes the core classes and dataflow?
 
I'm trying to understand how things fit together and work, and (perhaps due to my limited TypeScript knowledge) I'm having difficulty tracing things. I figure it might help if I knew which were the important classes and the main ways data flows around the internals.

For example, where should I look for the main classes which store the atom coordinates and biophysical structure information? How do those interact with the classes which control how those structures are rendered/displayed? There seems to be a distinction between trajectories/models/structures in the code -- how does mol* think about/distinguish these different items?

Is there any place such documentation might be already written down?

**Comment by dsehnal** [comment]
There is [this publication](https://diglib.eg.org/bitstream/handle/10.2312/molva20181103/029-033.pdf?sequence=1&isAllowed=y) which outlines the overall architecture. There is also some documentation [here](https://molstar.org/docs), but nothing as detailed as you are probably looking for.
